### PR TITLE
Rewrite by loader.Program instead of by ast.File

### PR DESCRIPTION
**Pull request by haya14busa**

### CODE REVIEW

File: cmd/gotypeconv/main.go [hunk 1]
```diff
@@ -49,25 +46,19 @@ func run(w io.Writer, args []string, opt *option) error {
...
+		return err
+	}
 	for _, pkg := range prog.InitialPackages() {
 		for _, f := range pkg.Files {
-			semaphore <- 1
-			eg.Go(func() error {
-				defer func() { <-semaphore }()
-				return runFile(w, opt, prog, pkg, f, typeErrs)
-			})
+			printFile(w, opt, prog, f)
```
Comment (review-dog) [inline cmd/gotypeconv/main.go]: **[errcheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
	printFile(w, opt, prog, f)

Comment (haya14busa) [reply cmd/gotypeconv/main.go]: Thank you reviewdog :joy:

File: cmd/gotypeconv/main_test.go [hunk 2]
```diff
@@ -2,9 +2,55 @@ package main
 
 import (
 	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
 	"testing"
+
+	ddiff "github.com/kylelemons/godebug/diff"
```
Comment (review-dog) [inline cmd/gotypeconv/main_test.go:11]: **[unused]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
could not import github.com/kylelemons/godebug/diff (cannot find package "github.com/kylelemons/godebug/diff" in any of:

File: cmd/gotypeconv/main_test.go [hunk 3]
```diff
@@ -2,9 +2,55 @@ package main
...
+		if err != nil {
+			t.Fatalf("%s: %v", fname, err)
+		}
+		defer gf.Close()
+		b, err := ioutil.ReadAll(gf)
+		if err != nil {
+			t.Fatalf("%s: %v", fname, err)
+		}
+
+		if d := ddiff.Diff(buf.String(), string(b)); d != "" {
```
Comment (review-dog) [inline cmd/gotypeconv/main_test.go:48]: **[unused]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
undeclared name: ddiff

File: cmd/gotypeconv/main_test.go (hunk 2 above)
Comment (review-dog) [inline cmd/gotypeconv/main_test.go:11]: **[gosimple]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
could not import github.com/kylelemons/godebug/diff (cannot find package "github.com/kylelemons/godebug/diff" in any of:

File: cmd/gotypeconv/main_test.go (hunk 3 above)
Comment (review-dog) [inline cmd/gotypeconv/main_test.go:48]: **[gosimple]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
undeclared name: ddiff

File: cmd/gotypeconv/main_test.go (hunk 2 above)
Comment (review-dog) [inline cmd/gotypeconv/main_test.go:11]: **[staticcheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
could not import github.com/kylelemons/godebug/diff (cannot find package "github.com/kylelemons/godebug/diff" in any of:

File: cmd/gotypeconv/main_test.go (hunk 3 above)
Comment (review-dog) [inline cmd/gotypeconv/main_test.go:48]: **[staticcheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
undeclared name: ddiff

File: cmd/gotypeconv/main_test.go (hunk 2 above)
Comment (haya14busa) [reply cmd/gotypeconv/main_test.go:11]: my bad... fixed 911b286

**Pull request merged**